Tasks/KubernetesManifestV0/src/CanaryDeploymentHelper.ts

```diff
@@ -0,0 +1,105 @@
+"use strict";
+import { Kubectl } from "utility-common/kubectl-object-model";
+import * as helper from './KubernetesObjectUtility';
+import * as utils from "./utilities";
+import path = require('path');
+import fs = require("fs");
+
+const BASELINE_SUFFIX = "-baseline";
+const BASELINE_LABEL_VALUE = "baseline";
+const CANARY_SUFFIX = "-canary";
+const CANARY_LABEL_VALUE = "canary";
+const CANARY_VERSION_LABEL = "azure-pipelines/version";
+
+export function calculateReplicaCountForCanary(inputObject: any, percentage: number) {
+    var inputReplicaCount = helper.getReplicaCount(inputObject);
+    return Math.round((inputReplicaCount * percentage) / 100);
+}
+
+export function isDeploymentEntity(kind: string): boolean {
+    if (!kind) {
+        throw new Error("Kind is not defined");
+    }
+
+    var temp = kind.toUpperCase();
+    return temp === helper.KubernetesWorkload.Pod.toUpperCase() ||
+        temp === helper.KubernetesWorkload.Replicaset.toUpperCase() ||
+        temp === helper.KubernetesWorkload.Deployment.toUpperCase() ||
+        temp === helper.KubernetesWorkload.StatefulSet.toUpperCase() ||
+        temp === helper.KubernetesWorkload.DaemonSet.toUpperCase();
+}
+
+export function getNewBaselineResource(stableObject: any, replicas: number): object {
+    return getNewCanaryObject(stableObject, replicas, BASELINE_LABEL_VALUE);
+}
+
+export function getNewCanaryResource(inputObject: any, replicas: number): object {
+    return getNewCanaryObject(inputObject, replicas, CANARY_LABEL_VALUE);
+}
+
+function getNewCanaryObject(inputObject: any, replicas: number, type: string): object {
+    var newObject = JSON.parse(JSON.stringify(inputObject));
+
+    // Updating name
+    newObject.metadata.name = type === CANARY_LABEL_VALUE ? getCanaryResourceName(inputObject.metadata.name) :
+        getBaselineResourceName(inputObject.metadata.name);
+
+    // Adding labels and annotations.
+    addCanaryLabelsAndAnnotations(newObject, type);
+
+    // Updating no. of replicas
+    if (newObject.kind.toUpperCase() != helper.KubernetesWorkload.Pod.toUpperCase() &&
+        newObject.kind.toUpperCase() != helper.KubernetesWorkload.DaemonSet.toUpperCase()) {
+        newObject.spec.replicas = replicas;
+    }
+
+    return newObject;
+}
+
+function addCanaryLabelsAndAnnotations(inputObject: any, type: string) {
+    var newLabels = new Map<string, string>();
+    newLabels[CANARY_VERSION_LABEL] = type;
+
+    helper.updateObjectLabels(inputObject, newLabels, false);
+    helper.updateObjectAnnotations(inputObject, newLabels, false);
+    helper.updateSelectorLabels(inputObject, newLabels, false);
+    helper.updateSpecLabels(inputObject, newLabels, false);
+}
+
+export function applyResource(kubectl: Kubectl, inputObjects: any[]) {
```

Comment:
You could move this function to `deploy.ts` instead?
And break it into a function which just creates those new files, and use that file list to the apply function being called there?